cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -1952,23 +1957,21 @@ func (p *BicepProvider) ensureParameters(
 		return nil, fmt.Errorf("resolving bicep parameters file: %w", err)
 	}
 
-	if len(template.Parameters) == 0 {
+	parametersOrderedKeys := template.Parameters.OrderedKeys()
+	parametersCount := len(parametersOrderedKeys)
+	if parametersCount == 0 {
 		return azure.ArmParameters{}, nil
 	}
-	configuredParameters := make(azure.ArmParameters, len(template.Parameters))
-
-	sortedKeys := maps.Keys(template.Parameters)
-	slices.Sort(sortedKeys)
-
+	configuredParameters := make(azure.ArmParameters, parametersCount)
```

Comment:
Can you provide more context here on the change? Is the old behavior that we'd read the parameters in an unordered fashion from the ARM JSON, and then impose our own ordering using sort? 

And we think the preservation of the ARM template JSON ordering is better?